packages/ember-glimmer/tests/integration/components/closure-components-test.js

```diff
@@ -0,0 +1,669 @@
+      {{#with (hash comp=(component "-looked-up" greeting=greeting)) as |my|}}
+        {{#my.comp}}{{/my.comp}}
+      {{/with}}`);
+
+    this.assertText('');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('');
+
+    this.runTask(() => this.context.set('greeting', 'Hodi'));
+
+    this.assertText('Hodi');
+
+    this.runTask(() => this.context.set('greeting', undefined));
+
+    this.assertText('');
+  }
+
+  ['@test nested components overwrite named positional parameters']() {
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['name', 'age']
+      }),
+      template: '{{name}} {{age}}'
+    });
+
+    this.render('{{component (component (component "-looked-up" "Sergio" 29) "Marvin" 21) "Hodari"}}');
+
+    this.assertText('Hodari 21');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hodari 21');
+  }
+
+  ['@test nested components overwrite hash parameters']() {
+    this.registerComponent('-looked-up', {
+      template: '{{greeting}} {{name}} {{age}}'
+    });
+
+    this.render(strip`
+      {{#with (component "-looked-up" greeting="Hola" name="Dolores" age=33) as |first|}}
+        {{#with (component first greeting="Hej" name="Sigmundur") as |second|}}
+          {{component second greeting=greeting}}
+        {{/with}}
+      {{/with}}`, {
+        greeting: 'Hodi'
+      });
+
+    this.assertText('Hodi Sigmundur 33');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hodi Sigmundur 33');
+
+    this.runTask(() => this.context.set('greeting', 'Kaixo'));
+
+    this.assertText('Kaixo Sigmundur 33');
+
+    this.runTask(() => this.context.set('greeting', 'Hodi'));
+
+    this.assertText('Hodi Sigmundur 33');
+  }
+
+  ['@skip bound outer named parameters get updated in the right scope']() {
+    this.registerComponent('-inner-component', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['comp']
+      }),
+      template: '{{component comp "Inner"}}'
+    });
+
+    this.registerComponent('-looked-up', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: ['name', 'age']
+      }),
+      template: '{{name}} {{age}}'
+    });
+
+    this.render('{{component "-inner-component" (component "-looked-up" outerName outerAge)}}', {
+      outerName: 'Outer',
+      outerAge: 28
+    });
+
+    this.assertText('Inner 28');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Inner 28');
+
+    this.runTask(() => this.context.set('outerAge', 29));
+
+    this.assertText('Inner 29');
+
+    this.runTask(() => this.context.set('outerName', 'Not outer'));
+
+    this.assertText('Inner 29');
+
+    this.runTask(() => {
+      this.context.set('outerAge', 28);
+      this.context.set('outerName', 'Outer');
+    });
+
+    this.assertText('Inner 28');
+  }
+
+  ['@skip bound outer hash parameters get updated in the right scope']() {
```

Comment:
Same answer. In other cases, we turn to commenting out the UR part, should I proceed with that?